### ISSUE TITLE
Bounding boxes drawn on training data

### ISSUE THREAD

**Issue by alanjschoen** [experiencor/basic-yolo-keras#55]
As I was debugging, I decided to look at some of the input images in training.  I noticed that bounding boxes and labels were drawn on them, and then I traced it to this bit of code in the `__getitem__` method in `BatchGenerator`:
```
            # assign input image to x_batch
            if self.norm != None: 
                x_batch[instance_count] = self.norm(img)
            else:
                # plot image and bounding boxes for sanity check
                for obj in all_objs:
                    if obj['xmax'] > obj['xmin'] and obj['ymax'] > obj['ymin']:
                        cv2.rectangle(img[:,:,::-1], (obj['xmin'],obj['ymin']), (obj['xmax'],obj['ymax']), (255,0,0), 3)
                        cv2.putText(img[:,:,::-1], obj['name'], 
                                    (obj['xmin']+2, obj['ymin']+12), 
                                    0, 1.2e-3 * img.shape[0], 
                                    (0,255,0), 2)
                        
                x_batch[instance_count] = img
```
So it looks to me that unless norm function is set, bounding boxes are drawn on the training data, and the norm function is not set in either train.py or the Step-By-Step notebook.  Is this the desired behavior?

**Comment by experiencor** [comment]
@alanjschoen The norm is the normalization function, which is different for different backends. I set it in frontend.py, when creating the train generator and validation generator (lines 403 and 406).

**Comment by alanjschoen** [comment]
@experiencor I looked at frontend.py and I see what youre talking about, so this should not affect train.py.  But this problem might still affect the Step-By-Step notebook because it creates the BatchGenerator objects directly instead letting the `train` function in frontend.py create them, and it does not set the norm function.

**Comment by experiencor** [comment]
@alanjschoen You are right. My bad. I did not change the code in the notebook accordingly.

**Comment by experiencor** [comment]
I changed the notebook to work with the new preprocessing.py.

**Comment by letilessa** [comment]
Why is the normalization function different for different backends? In particular, why mobilenet's normalize is the following?

`def normalize(self, image):
        image = image / 255.
        image = image - 0.5
        image = image * 2.

        return image`

**Comment by rodrigo2019** [comment]
Because it fallows the original normalizations for each model.
In the mobileNet you can see the original implementation [here](https://github.com/tensorflow/models/blob/master/research/slim/nets/mobilenet/mobilenet_example.ipynb)

> images = tf.cast(images, tf.float32) / 128.  - 1

 and the keras implementation [here](https://github.com/fchollet/deep-learning-models/blob/master/mobilenet.py#L91). The formulas are quite different, but gives very close results.

And by default when the model doesnt specific the normalization, it is used 1/255 that in theory converges more quickly than using values from 0..255
